### PR TITLE
1) Migration to TypeScript (if we DON'T NEED to track files content history)

### PR DESCRIPTION
**Pull request by MaksimkaSota**
Feature:

- Added TypeScript in the project;
- Added TypeScript settings for ESLint/Prettier in the project.

Refactor:

- Migrating files from `.js` to `.ts`/`.tsx`;
- Code style for all project;

When a file has been renamed and changed in one commit, GitHub displays it as one deleted file and one added file.
This means that a multiline file will be reviewed without context!

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -87,6 +87,17 @@
...
+    ],
+
+    "@typescript-eslint/default-param-last": "off",
+    "@typescript-eslint/consistent-type-imports": "error",
+    "@typescript-eslint/no-restricted-imports": [
+      "warn",
+      {
+        "name": "react-redux",
+        "importNames": ["useSelector", "useDispatch"],
+        "message": "Use typed hooks 'useTypedDispatch' and 'useTypedSelector' instead."
```
Comment (ZhekaVasil) [inline .eslintrc.json:99]: Nice =)

File: src/utils/types/common.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import type { Action } from 'redux';
+import type { ThunkAction } from 'redux-thunk';
+import type { AppState } from '../../redux/reducers/reducers';
+
+export type Nullable<T> = T | null;
```
Comment (ZhekaVasil) [inline src/utils/types/common.ts:5]: 👍🏾

File: src/Components/Main/Pages/NotFoundPage/NotFoundPage.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
+import type { FC, ReactElement } from 'react';
 import { useLocation } from 'react-router-dom';
 import { Error } from '../../../Common/Error/Error';
 
-export const NotFoundPage = () => {
+export const NotFoundPage: FC = (): ReactElement => {
   const { pathname } = useLocation();
-  const message = `The requested URL "${pathname}" was not found on this server!`;
+  const message: string = `The requested URL "${pathname}" was not found on this server!`;
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/NotFoundPage/NotFoundPage.tsx]: As I know, in such obvious string cases - it is not  required to set string type =)

File: src/Components/Main/Pages/UsersPage/UsersPageContainer.tsx [hunk 4]
```diff
@@ -14,27 +14,29 @@ import {
...
-  const users = useSelector(usersSelector);
-  const page = useSelector(pageSelector);
-  const pageSize = useSelector(pageSizeSelector);
-  const totalCount = useSelector(totalCountSelector);
-  const subscribersId = useSelector(subscribersIdSelector);
-  const isFetchingUsers = useSelector(isFetchingUsersSelector);
-  const usersError = useSelector(usersErrorSelector);
-  const followErrors = useSelector(followErrorsSelector);
-  const unfollowErrors = useSelector(unfollowErrorsSelector);
+const UsersPageContainer: FC = (): ReactElement | boolean => {
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/UsersPage/UsersPageContainer.tsx:20]: image

File: src/hooks/useMounted.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { useEffect, useRef } from 'react';
 
-export const useMounted = () => {
-  const isMounted = useRef(false);
+export const useMounted = (): boolean => {
+  const isMounted = useRef<boolean>(false);
```
Comment (ZhekaVasil) [inline src/hooks/useMounted.ts]: It is also not required to set ```<boolean>``` type here, you provided default value and TS with use type from the default value =)

File: src/redux/store.ts [hunk 6]
```diff
@@ -2,5 +2,6 @@ import { applyMiddleware, compose, legacy_createStore as createStore } from 'red
 import thunkMiddleware from 'redux-thunk';
 import { rootReducer } from './reducers/reducers';
 
+// @ts-ignore
```
Comment (ZhekaVasil) [inline src/redux/store.ts]: MAy be helpful =)
https://stackoverflow.com/questions/12709074/how-do-you-explicitly-set-a-new-property-on-window-in-typescript

File: .eslintrc.json (hunk 1 above)
Comment (MaksimkaSota) [reply .eslintrc.json:99]: Thanks :)

File: src/utils/types/common.ts (hunk 2 above)
Comment (MaksimkaSota) [reply src/utils/types/common.ts:5]: 👍🏻

File: src/Components/Main/Pages/NotFoundPage/NotFoundPage.tsx (hunk 3 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/NotFoundPage/NotFoundPage.tsx]: It is a redundant code. I have fixed it!

File: src/hooks/useMounted.ts (hunk 5 above)
Comment (MaksimkaSota) [reply src/hooks/useMounted.ts]: It is a redundant code. I have fixed it!

File: src/Components/Main/Pages/UsersPage/UsersPageContainer.tsx (hunk 4 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/UsersPage/UsersPageContainer.tsx:20]: I do not have this problem...  

image

File: src/redux/store.ts (hunk 6 above)
Comment (MaksimkaSota) [reply src/redux/store.ts]: Thanks, this was helpful :)
Done!

**Comment by MaksimkaSota** [comment]
Oops... I have used "--force" for rename last commit :)

**Review by ZhekaVasil — APPROVED** [review]

**Pull request merged**